Update to 1.20.4 (Must be tested before merging)

TechPro424 opened the pull request:
Updates the mod to 1.20.4

TechPro424 committed:
Add remappedSrc to .gitignore

TechPro424 committed:
Update Loom and Gradle version

TechPro424 committed:
Update gradle.properties

TechPro424 committed:
Fixed an import

TechPro424 committed:
An extremely half-baked fix for the reconnect override

Needs to be tested

The method signature for the IntegratedServerLoader.start() method was changed in either 1.20.3 or 1.20.4
It now needs a runnable as the 2nd parameter (parameter name is onCancel)
So as of now the onCancel runnable is just a lambda function as I am not sure what the runnable should do (maybe return the player to the title screen?)
The runnable seems to be run whenever an exception is encountered while running the start() method and the world cannot be "recovered" (not too sure what that means)

TechPro424 committed:
Edited comment to add more info

TechPro424 committed:
Fixed an import

TechPro424 committed:
Extremely half-baked fix for the reconnect override

needs to be tested

The RealmsGetServerDetailsTask seems to have been removed in either 1.20.3 or 1.20.4
The 2 Tasks that seemed to fit the most were OpenServerTask and RealmsPrepareConnectionTask
I looked into the OpenServerTask, and saw that if the boolean join was true, it would execute a play method in the RealmsMainScreen class that in turn runs the RealmsPrepareConnectionTask
Hence I decided to use the OpenServerTask for now

TechPro424 committed:
Fixed the MinecraftClientMixin
By fixing an inject to the startIntegratedServer method
The method signature had changed

itsTyrion commented:
you can just check what the game does, there are 2 linked functions in the jdoc comments (`@see net.minecraft.client.QuickPlay#startRealms`)
correct, the lambda functions just set the screen in this case.
I made it cleaner in my PR (that I did without checking if there already is one xD)

TechPro424 committed:
Make the lamba function in the SingleplayerReconnectStrategy send the player to the title screen

Just found out that the onCancel lambda sends the player to the title screen, so made the mod replicate that
Thanks to itsTyrion on GitHub for pointing that out!
(See https://github.com/Bstn1802/AutoReconnect/pull/65#issuecomment-1938891924)

TechPro424 commented:
> you can just check what the game does, there are 2 linked functions in the jdoc comments (`@see net.minecraft.client.QuickPlay#startRealms`) correct, the lambda functions just set the screen in this case. I made it cleaner in my PR (that I did without checking if there already is one xD)

Thanks for pointing that out! Since there are 2 PRs, do you want me to close mine or are you ok with closing yours?

(Also I think you meant startSingleplayer there instead of startRealms)

TechPro424 committed:
Make the reconnectMethod override use the RealmsPrepareConnectionTask, replicating the vanilla game
Also removed the comments warning about the untested changes

TechPro424 commented:
Also fixed the reconnectMethod override in RealmsReconnectStrategy
